File: fsharpfunctions/DataProcessing/IssueData.fs
namespace FSharpFunctions
open System

type EntryType = CountOnly | ByPriority | ByService | BySubService

type Source = CSharp | FSharp

(*
    option = Some<T> | None
*)

type IssueData = {
    id: string
    Source: Source
    EntryType: EntryType //we know what possible values are
    Timestamp: DateTime
    TotalOpenIssues: int
    MissingTags: int
    CountByPriority: option<Map<string, int>> //immutable key-value store
    CountByService: option<Map<string, int>>
}

File: csharpfunctions/DataProcessing/IssueData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace csharpfunctions
{
    public class IssueData
    {
        public string id { get; set; }
        public string Source { get; set; }
        public string EntryType { get; set; }
        public DateTime Timestamp { get; set; }
        public int TotalOpenIssues { get; set; }
        public int MissingTags { get; set; }
        public Dictionary<string, int> CountByPriority { get; set; }
        public Dictionary<string, int> CountByService { get; set; }
    }
}
